zinit-install.zsh

```diff
@@ -2229,19 +2262,20 @@ zimv() {
         local dir="${5#%}" hook="$6" subtype="$7" || \
         local dir="${4#%}" hook="$5" subtype="$6"
 
-    if [[ ( $hook = *\!at(clone|pull)* && ${+ICE[nocompile]} -eq 0 ) || \
-            ( $hook = at(clone|pull)* && $ICE[nocompile] = '!' )
-    ]] {
-        # Compile plugin
-        if [[ -z $ICE[(i)(\!|)(sh|bash|ksh|csh)] ]] {
-            () {
-                emulate -LR zsh
-                setopt extendedglob warncreateglobal
-                if [[ $tpe == snippet ]] {
-                    .zinit-compile-plugin "%$dir" ""
-                } else {
-                    .zinit-compile-plugin "$id_as" ""
-                }
+    if ! [[ ( $hook = *\!at(clone|pull)* && ${+ICE[nocompile]} -eq 0 ) || \
+            ( $hook = at(clone|pull)* && $ICE[nocompile] = '!' ) ]] {
+        return 0
+    }
+
+    # Compile plugin
+    if [[ -z $ICE[(i)(\!|)(sh|bash|ksh|csh)] ]] {
```

Comment:
No idea what `$ICE[(i)(\!|)(sh|bash|ksh|csh)` evaluates to

Reply:
This looks for and returns the first key (ie `sh`, `bash`, `ksh` etc) of the `ICE`map that matches `(sh|bash|ksh|csh)` and excludes `!bash`, `!sh` etc.

Reply:
ie this check if any of the `sh`, `bash`, `ksh` or `csh` ices is set.

Reply:
Insane. Thanks for the explanation.

Reply:
Got it wrong btw. I've updated my comments.

Reply:
```zsh
typeset -A MYICE=([\!a]=valA [\!b]=valB [c]=C)
echo $MYICE[(i)(\!|)(a|b|c)]
!a
```